app/controllers/api/v1/accounts/conversations_controller.rb

```diff
@@ -89,6 +89,12 @@ def custom_attributes
     @conversation.save!
   end
 
+  def patch_custom_attributes
+    custom_attributes = params.permit(custom_attributes: {})[:custom_attributes]
+    sanitized_json = custom_attributes.to_json.gsub("'", "''")
+    Current.account.conversations.where(id: @conversation.id).update_all("custom_attributes = custom_attributes || '#{sanitized_json}'::jsonb")
```

Comment:
@nikhilmuz We already have `@conversation` available.  so the `update_all` doesn't look the right way to go about this. 
lets do something like 

```
@conversation.custom_attributes = (@conversation.custom_attributes || {}).deep_merge(params[:custom_attributes].stringify_keys)
@conversation.save!
```

ref: https://github.com/chatwoot/chatwoot/blob/71c5a1e1d4d46170abf4d19c708cc3cba745a949/app/actions/contact_identify_action.rb#L129

Reply:
The idea was to avoid race condition so doing it on the ruby layer doesn't seems right to me. Basically, it fetches conversation somewhere in the controller and we do a deep merge with a new object and reinsert it overwriting all the existing data on parallel API calls. This way we are ensuring lock on the Postgres layer.

Reply:
@sojan-official

Reply:
@nikhilmuz We want the callbacks to happen on custom attribute updates for automation/webhooks to be triggered.